opm/common/data/SimulationDataContainer.cpp

```diff
@@ -75,6 +75,36 @@ namespace Opm {
     }
 
 
+    void SimulationDataContainer::setCellDataComponent( const std::string& key , size_t component , const std::vector<int>& cells , const std::vector<double>& values) {
+        auto& data = getCellData( key );
+        if (component >= m_num_phases)
```

Comment:
Ehhh - here I disagree; the restriction `if component < m_num_phases` will be retained, although variable names++ might change, whereas all remnants of `data.size() == m_num_phases * m_num_cells` should be eradicted completely in the future. I therefor actually feel it makes sense to keep them apart?!